Block/System/Config/Form/AbstractDefaultDropOffPoint.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Magento\Block\System\Config\Form;
+
+use Magento\Backend\Block\Template\Context;
+use Magento\Backend\Block\Widget\Button;
+use Magento\Config\Block\System\Config\Form\Field;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Data\Form\Element\AbstractElement;
+use MyParcelNL\Magento\Helper\Data;
+use MyParcelNL\Sdk\src\Model\Carrier\CarrierFactory;
+
+abstract class AbstractDefaultDropOffPoint extends Field
+{
+    /**
+     * Path to template file in theme.
+     *
+     * @var string
+     */
+    protected $_template = 'MyParcelNL_Magento::default_drop_off_point.phtml';
+
+    /**
+     * @var \MyParcelNL\Sdk\src\Model\Consignment\DropOffPoint
+     */
+    private $dropOffPoint;
+
+    /**
+     * @throws \Exception
+     */
+    public function __construct(Context $context, array $data = [])
+    {
+        $objectManager      = ObjectManager::getInstance();
+        $helper             = $objectManager->get(Data::class);
+        $dropOffPoint       = $helper->getDropOffPoint(CarrierFactory::createFromId($this->getCarrierId()));
+        $this->dropOffPoint = $dropOffPoint;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * @return int
+     */
+    abstract public function getCarrierId(): int;
+
+    /**
+     * Retrieve HTML markup for given form element
+     *
+     * @param  \Magento\Framework\Data\Form\Element\AbstractElement $element
+     *
+     * @return string
+     */
+    public function render(AbstractElement $element): string
+    {
+        $element->unsScope()
+            ->unsCanUseWebsiteValue()
+            ->unsCanUseDefaultValue();
+        return parent::render($element);
+    }
+
+    /**
+     * Retrieve element HTML markup
+     *
+     * @param  \Magento\Framework\Data\Form\Element\AbstractElement $element
+     *
+     * @return string
+     */
+    protected function _getElementHtml(AbstractElement $element): string
```

Comment:
kan deze param weg of gaat phpstorm dan miepen?

Reply:
Jaa dat vindt hij niet leuk..